sdk/auth/token.go

```diff
@@ -4,15 +4,54 @@
 package auth
 
 import (
-	"golang.org/x/oauth2"
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
 	"time"
 
 	"github.com/hashicorp/go-azure-sdk/sdk/claims"
+	"golang.org/x/oauth2"
 )
 
+// SetAuthHeaders decorates a *http.Request with necessary authorization headers for Azure APIs. For more information about the vendor-specific
+// `x-ms-authorization-auxiliary` header, see https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/authenticate-multi-tenant
+func SetAuthHeaders(ctx context.Context, req *http.Request, authorizer Authorizer) error {
```

Comment:
since this is specific to Resource Manager, this probably wants to go into the Resource Manager Client?

Reply:
agreed, other APIs may support the concept of multiple tenants but the implementation will be different